cmd/internal/pluginutils/bypath.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginutils
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+)
+
+// Returns a slice of by-path Mounts for a pci device with pciAddress.
+// by-path files are searched from the given bypathDir.
+// In the by-path dir, any files that start with "pci-<pci addr>" will be added to mounts.
+func BypathMountsForPci(pciAddress, bypathDir string) []pluginapi.Mount {
```

Comment:
could it be done inside the GPU plugin?

Reply:
Move bypath.go contents to gpu-plugin.go?

Reply:
I moved everything into gpu_plugin.go and combined the two bypath functions into one. Tests also.